RiotSwiftUI/Modules/UserSessions/UserSessionOverview/View/UserSessionOverviewToggleCell.swift

```diff
@@ -0,0 +1,86 @@
+//
+// Copyright 2022 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import SwiftUI
+
+struct UserSessionOverviewToggleCell: View {
+    @Environment(\.theme) private var theme: ThemeSwiftUI
+    
+    let title: String
+    let message: String?
+    let isOn: Bool
+    let isEnabled: Bool
+    var onBackgroundTap: (() -> Void)?
+    
+    var body: some View {
+        VStack(spacing: 8) {
+            Button(action: {
+                guard isEnabled else { return }
+                onBackgroundTap?()
+            }) {
+                VStack(spacing: 0) {
+                    SeparatorLine()
+                    Toggle(isOn: .constant(isOn)) {
```

Comment:
Looking at the animation on this, the toggle fades out on touch down. Should this instead using an `@State` for `isOn` and an `.onChange(of: isOn) { onToggleChange?($0) }` modifier instead of the button and `onBackgroundTap` closure?

Reply:
I would keep it like this for these 2 reasons if you don't mind:
- I'd like to have the same fade out effect between `Session's detail` cell and this cell
- This way the toggle changes only after having requested the change on the server.

Reply:
Sure 👍